flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdRowCount.scala

```diff
@@ -335,10 +336,21 @@ class FlinkRelMdRowCount private extends MetadataHandler[BuiltInMetadata.RowCoun
       fmq.getSelectivity(joinWithOnlyEquiPred, nonEquiPred)
     }
 
+    // Dynamic partition pruning factor is adding to adjust join cost for these join
+    // node which meets dynamic partition pruning pattern. Try best to reorder the
+    // fact table and fact table together to make dpp succeed.
```

Comment:
Currently, join-reorder is before dynamic partition pruning rewrite. This factor is adding to adjust join cost for these join node which meets dynamic partition pruning pattern. Try best to reorder the fact table and fact table together to make DPP succeed.

Reply:
> Currently, join-reorder is before dynamic partition pruning rewrite. This factor is adding to adjust join cost for these join node which meets dynamic partition pruning pattern. Try best to reorder the fact table and fact table together to make DPP succeed.

done!